packages/components/src/button/index.js

```diff
@@ -92,6 +92,12 @@ export function Button( props, ref ) {
 		'components-button__description'
 	);
 
+	const hasChildren =
+		children?.[ 0 ] &&
+		children[ 0 ] !== null &&
+		// Tooltip should not considered as a child
+		children?.[ 0 ]?.props?.className !== 'components-tooltip';
```

Comment:
If `Tooltip` is not excluded as a child, it will be considered to have children when moused over and `has-text` class will be given.
I couldn't find a good approach to not regard `Tooltip` as a child.
Is there a better way?

Reply:
> Otherwise, we could perhaps add an official API to Button that allows consumers to set some kind of "ignore me as a visible child" flag in CSS. So like if you add a designated CSS class, e.g. <Button><div className="ignore-as-components-button-child" /></Button>, Button could exclude it from its hasChildren check.

Perhaps I think this new API should be considered in the overall Button component in #44042 for consideration.

> Another way might be to only apply .has-text when the child is actually a ReactText. I'm not sure, but I'm guessing that there are not a lot of use cases that involve a icon prop and an element child with text.

Is it correct that a use case with a icon prop and an element child with text is something like the following?

```javascript
<Button icon={ help } variant="primary">
	Push Me
</Button>
```

I've seen such usage many times in my experience, so I think that the `has-text` class should be added.

> Long term, I'm hoping it could be resolved by the Tooltip rewrite. Maybe that will remove the need to inject direct children?

Yes, that's the challenge I must address in #42753 😅

Reply:
> > Another way might be to only apply .has-text when the child is actually a ReactText. I'm not sure, but I'm guessing that there are not a lot of use cases that involve a icon prop and an element child with text.
> 
> Is it correct that a use case with a icon prop and an element child with text is something like the following?

Sorry, this was unclear. By "icon prop and an element child with text" I meant something like:

```jsx
// *Element* child with text
<Button icon={ help }>
	<span>Push Me</span>
</Button>
```

in contrast to a text child:

```jsx
// *Text* child
<Button icon={ help }>
	Push Me
</Button>
```

Reply:
Thank you, I understand.
From what I have found on [GitHub Code Search](https://cs.github.com/?q=%40wordpress%2Fcomponent+%3E%3C%2FButton%3E), it appears that "an element child with text" is almost never used.
For now, however, I think it would have less impact to consider both "element child with text" and "text child" as "having text".

This specification may need to be reconsidered in relation to #44042, but I think it is reasonable in this PR.
If it is not a problem, I would like to merge it 🚀

Reply:
Yep, sounds good! 🚢